azure-kusto-data/azure/kusto/data/__init__.py

```diff
@@ -1,4 +1,148 @@
-from .kusto_client import KustoClient, KustoResultIter, KustoResponse
-from .kusto_exceptions import KustoError, KustoClientError, KustoServiceError
+from enum import Enum
+from six import text_type
 from .version import VERSION
 __version__ = VERSION
+
+class KustoConnectionStringBuilder(dict):
+    class _Keywords(Enum):
+        data_source = "Data Source"
+        aad_user_id = "AAD User ID"
+        password = "Password"
+        application_client_id = "Application Client Id"
+        application_key = "Application Key"
+        authority_id = "Authority Id"
+
+    def __init__(self, connection_string, *args, **kwargs):
+        self._constructing = True
+        if connection_string is not None and '=' not in connection_string.split(';')[0]:
```

Comment:
Is the empty string acceptable as a connection_string? Are you really expecting people to pass in `None`? If the answer is "no" to any of this then that guard either needs to become `if connection_string` or simply drop the guard.
Use `connection_string.partition(';')[0]` instead of `split()`.